diff -up cython-0.29.28/Cython/Compiler/ModuleNode.py.omv~ cython-0.29.28/Cython/Compiler/ModuleNode.py
--- cython-0.29.28/Cython/Compiler/ModuleNode.py.omv~	2022-03-19 18:48:56.463782075 +0100
+++ cython-0.29.28/Cython/Compiler/ModuleNode.py	2022-03-19 18:57:31.325762579 +0100
@@ -642,6 +642,7 @@ class ModuleNode(Nodes.Node, Nodes.Block
         for inc in sorted(env.c_includes.values(), key=IncludeCode.sortkey):
             if inc.location == inc.INITIAL:
                 inc.write(code)
+        code.putln('#include "internal/pycore_frame.h"') # Needed for PyFrameObject in Python 3.11a6, PyFrameObject is used heavily by cython
         code.putln("#ifndef Py_PYTHON_H")
         code.putln("    #error Python headers needed to compile C extensions, "
                    "please install development version of Python.")
